doc/_extensions/zephyr/domain.py

```diff
@@ -111,6 +111,30 @@ def convert_node(self, node):
             for sibling in siblings_to_move:
                 parent.remove(sibling)
 
+            # Set sample description as the meta description of the document for improved SEO
+            meta_description = nodes.meta()
+            meta_description["name"] = "description"
+            meta_description["content"] = node.children[0].astext()
+            node.document += meta_description
+
+            # Similarly, add a node with JSON-LD markup (only renders in HTML output) describing
+            # the code sample.
+            json_ld = nodes.raw(
+                "",
+                f"""<script type="application/ld+json">
+                {json.dumps({
+                    "@context": "http://schema.org",
+                    "@type": "SoftwareSourceCode",
+                    "name": node['name'],
+                    "description": node.children[0].astext(),
+                    "codeSampleType": "full",
+                    "codeRepository": vcs_link_get_url(self.app, self.env.docname)
+                })}
+                </script>""",
+                format="html",
+            )
+            node.document += json_ld
```

Comment:
question: is this inserted regardless of the renderer? ie, what happens for PDF or any others?

Reply:
see the comment at line 120 -- since the format of the raw node is set to HTML, this should not be in e.g. Latex output. I will actually check that's the case but see for example https://github.com/sphinx-doc/sphinx/blob/00ea93d05116312edea40d2678480df32b270e94/sphinx/writers/latex.py#L1729-L1737

Reply:
FWIW I did end up checking and... forgot that the code samples are not mounted in the PDF document anyway :)